synapse/module_api/__init__.py

```diff
@@ -1230,6 +1230,18 @@ def looping_background_call(
                 f,
             )
 
+    def should_run_background_tasks(self) -> bool:
```

Comment:
This seems like a fairly low-level thing to expose -- can you expand a bit for why we need this? Is this to be paired with #15993 inside a module? Would it make more sense to have the same `run_on_all_instances` that `looping_background_call` has?

Reply:
It's to be paired with `looping_background_call` for registering an event handler that may be useful in conjunction with the background call.
I get it's a bit weird but the alternative is having every module need to accept its own configuration for which worker to use as the background worker (often we do this and default to the master process).

(Maybe I should just push the configuration and logic up to the module and use `run_on_all_instances=True` but only register the background call on one worker?)

Reply:
> (Maybe I should just push the configuration and logic up to the module and use `run_on_all_instances=True` but only register the background call on one worker?)

I'm confused -- why can't you just let it do `run_on_all_instances=False`? It sounds like the goal is to run it on a single worker?

Reply:
I need to register an event handler for new events on the same worker as the one running the background jobs.